src/dstore/pmix_esh.c

```diff
@@ -574,62 +863,96 @@ static int _esh_patch_env(char ***env)
     return pmix_setenv(PMIX_DSTORE_ESH_BASE_PATH, _base_path, true, env);
 }
 
-static int _esh_nspace(const char *nspace)
+static int _esh_nspace(const char *nspace, pmix_info_t info[], size_t ninfo)
 {
     struct stat st = {0};
     seg_desc_t *seg;
+    int rc;
+    int tbl_idx;
+    nspace_tbl_entry_t *nspace_tbl_entry = NULL;
```

Comment:
Lets add one more info key here for the future use: session ID. If this key was passed to use we construct the path in the following way:
`<tmpdir_path>/<dstor-root-dir>/<uid>/<session-id>`
otherwise the path will be just 
`<tmpdir_path>/<dstor-root-dir>/<uid>/`
This will help when in future there will be multi-session implementations.